### ISSUE TITLE
packaged artifact names changed in 1.0.0-RC3

### ISSUE THREAD

**Issue by muuki88** [sbt/sbt#3388]
## steps

* checkout sbt-native-packager on branch `wip/cross-build-sbt-1.0`
* Set version `; ++2.12.2 ; ^^1.0.0-RC3`
* Run `scripted jar/*` tests

## problem

The jar file names should contain the `organization`, but it is missing.
All tests in the `jar/*` folder check the `MANIFEST.MF` that the classpath
has all necessary jars.

[The error  message](https://travis-ci.org/sbt/sbt-native-packager/jobs/258851351#L1502) indicates that the names have changed:

```
java.lang.AssertionError: assertion failed: 
MANIFEST Class-Path should contain com.typesafe.config:
[info] [error] launcher-jar-test.launcher-jar-test-0.1.0.jar launcher-jar-test_2.12-0.1.0.jar scala-library.jar config-1.3.1.jar
```

Instead I have `config-1.3.1.jar`.

( the actual log shows 'akka-actor', which is simply wrong output. I'm fixing that in the next commits )

## expectation

The exported jars contain the `organization` and `normalizedName`.

## notes

sbt version: *1.0.0-RC3*

**Comment by eed3si9n** [comment]
This is 0.13's artifactName https://github.com/sbt/sbt-zero-thirteen/blob/0.13/ivy/src/main/scala/sbt/Artifact.scala#L80-L87:

```scala
 def artifactName(scalaVersion: ScalaVersion, module: ModuleID, artifact: Artifact): String =
    {
      import artifact._
      val classifierStr = classifier match { case None => ""; case Some(c) => "-" + c }
      val cross = CrossVersion(module.crossVersion, scalaVersion.full, scalaVersion.binary)
      val base = CrossVersion.applyCross(artifact.name, cross)
      base + "-" + module.revision + classifierStr + "." + artifact.extension
    }
```

**Comment by muuki88** [comment]
Thanks @eed3si9n :)

I found the responsible lines in native-packager that cause this regression:
https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppPackaging.scala#L103-L109

```scala
private def getJarFullFilename(dep: Attributed[File]): String = {
  val filename: Option[String] = for {
    module <- dep.metadata.get(AttributeKey[ModuleID]("module-id"))
    artifact <- dep.metadata.get(AttributeKey[Artifact]("artifact"))
  } yield makeJarName(module.organization, module.name, module.revision, artifact.name, artifact.classifier)
  filename.getOrElse(dep.data.getName)
}
```

Both `dep.metadata` calls return `None` and the `dep.data.getName` is used as a fallback, which causes the failing tests. I'll take a deeper look on how to fix this for both sbt 0.13.x and 1.x.

**Comment by muuki88** [comment]
Turns out that the `module-id` key was renamed to `moduleID` which causes the for-comprehension to return `None`.
